app/src/main/java/ch/sdp/vibester/database/DataGetter.kt

```diff
@@ -79,18 +79,37 @@ class DataGetter @Inject constructor() {
     fun updateSubFieldInt(userID: String, newVal: Int, fieldName: String, subFieldName: String, method: String) {
         dbUserRef.child(userID).child(fieldName).child(subFieldName)
             .get().addOnSuccessListener { t ->
-                var finalVal = newVal
-                if(t.value != null) {
+                var finalVal = checkValue(t, method, newVal) //newVal: TO DELETE IF TESTS PASS, OTHERWISE ROLL BACK!
+                /*if(t.value != null) {
                     val previousVal = (t.value as Long?)!!.toInt()
                     when (method) {
                         "sum" -> finalVal += previousVal
                         "best" -> finalVal = maxOf(previousVal, newVal)
                     }
-                }
+                }*/
                 setSubFieldValue(userID, fieldName, subFieldName, finalVal)
             }
     }
 
+    /**
+     * Depending on the value of the snapshot and the method with which we change the value,
+     * returns either the sum or the best value for the finalVal.
+     * @param t         : DataSnapshot
+     * @param method    : String, mode with which we assign finalVal
+     * @param newVal    : Int, value to compare to previousVal
+     */
+    private fun checkValue(t: DataSnapshot, method: String, newVal: Int): Int {
+        var finalVal = newVal
+        if(t.value != null) {
+            val previousVal = (t.value as Long?)!!.toInt()
+            when (method) {
+                "sum" -> finalVal += previousVal
+                "best" -> finalVal = maxOf(previousVal, newVal)
+            }
+        }
+        return finalVal
+    }
+
```

Comment:
Good to have refactored that into a separate function, it avoids duplications and lengthy functions. 👍